src/app/core/messages.service.ts

```diff
@@ -24,11 +24,10 @@ export class MessagesService {
                     catchError(this.handleError)
                 )
                 .toPromise()
-                .then((val: any) => {
-                    if (val && val.length) {
-                        this._message = new Message(val[0]['_id'], 'error', val[0]['maintenanceMessage']);
-                        return this._message;
-                    } else { throw new Error('System status returned a malformed message.'); }
+                .then((val: Message) => {
+                    // tslint:disable-next-line: no-use-before-declare
```

Comment:
Can we remove this comment and move the below `Message` class to the top of the file?